react/week4/searcher/src/SearchContext.js

```diff
@@ -0,0 +1,67 @@
+import { createContext, useEffect, useState } from "react";
+
+export const SearchContext = createContext();
+const API_URL = "https://api.github.com/search/users?q=";
+const HYF_REPOS_URL = "https://api.github.com/users/HackYourFuture-CPH/repos";
+
+const SearchProvider = ({ children }) => {
+  const [query, setQuery] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+  const [users, setUsers] = useState([]);
+  const [hyfRepos, setHyfRepos] = useState([]);
+
+  useEffect(() => {
+    if (query === "") {
+      setUsers([]);
+      return;
+    }
+
+    setIsLoading(true);
+    fetch(API_URL + query)
+        .then(response => {
+            if (response.ok) {
+                return response.json();
+            } else {
+                throw new Error(`Error code ${response.status}`);
+            }
+        })
+        .catch(er => {
+            setError(er.toString());
```

Comment:
Remember to reset your error, either when the search is successful or when the input is empty.